draft-ietf-httpbis-priority.md

```diff
@@ -509,8 +509,54 @@ not specify the incremental(`i`) parameter.
 
 # Client Scheduling
 
-A client MAY use priority values to make local scheduling choices about the
-requests it initiates.
+A client MAY use priority values to make local processing or scheduling choices
+about the requests it initiates.
+
+
+# Server Scheduling
+
+Priority signals are input to a prioritization process. They do not guarantee
+any particular processing or transmission order for one response relative to any
+other response. An endpoint cannot force a peer to process concurrent request in
+a particular order using priority. Expressing priority is therefore only a
+suggestion.
+
+A server can use Priority signals along with other inputs to make scheduling
+decisions. No guidance is provided about how this can or should be done. Factors
+such as implementation choices or deployment environment also play a role. Any
+given connection is likely to have many dynamic permutations. For these reasons,
+there is no unilateral perfect scheduler and this document only provides some
+basic recommendations for implementations. 
+
+Clients can expect servers will make prioritization decisions, including
+ignoring all signals. And they should expect that decisions might be based on
+metadata or information beyond the scope of extensible priorities. 
+
+It is RECOMMENDED that, when possible, servers respect urgency values. Sending
+higher urgency responses before lower urgency responses.
+
+It is RECOMMENDED that, when possible, servers respect incremental values.
+Sending non-incremental responses of the same urgency one-by-one, and sending
+incremental responses of the same urgency in round-robin manner.
+
+It is RECOMMENDED that a server considers request generation order as an input
+to prioritization. This can be determined from the Stream ID. A server that
+receives concurrent requests at the same urgency level might serve the responses
+one-by-one but it needs to pick an order. Serving the lowest Stream ID in a
+given urgency level can align well with clients usage of HTTP; such as user
+agents that load document trees where ordering is important.
+
+For non-incremental resources the total download time (time to first byte - time
```

Comment:
isn't it time from request to last byte delivered?

Reply:
This is probably open to interpretation a little so it would be good to hammer it down. Looking at this again, I don't like how I presented it and will tweak it. I was basing it on my understanding of curl and chrome as shown here https://blog.cloudflare.com/a-question-of-timing/. Do other client have alternative views?

My thinking, based on other discussions, is that a non-incremental payload can only be used when the whole payload is received. Therefore, factoring in the delta between request and TTFB is not super helpful in this context. However, non-incremental objects probably do benefit from a shorter TTFB and "time to significant delineator", such as a progressive jpeg header.